### PR TITLE
fix(tree): allow deselection in single selectionMode

### PR DESCRIPTION
**Pull request by josercarcamo**
**Related Issue:** #7900

## Summary
Made sure an item can be selected and deselected in single selection mode.

### CODE REVIEW

**Review by driskull — APPROVED** [review]
Code looks good. For the PR title, maybe:

`fix(tree): allow deselection in single selectionMode`

or

``fix(tree): support unselecting in single selectionMode``

File: packages/calcite-components/src/components/tree/tree.tsx [hunk 1]
```diff
@@ -228,7 +228,7 @@ export class Tree {
     } else if (!isNoneSelectionMode) {
       targetItems.forEach((treeItem) => {
         if (!treeItem.disabled) {
-          treeItem.selected = true;
+          treeItem.selected = this.selectionMode === "single" ? !treeItem.selected : true;
```
Comment (jcfranco) [inline packages/calcite-components/src/components/tree/tree.tsx:231]: Suggestion: `this.selectionMode === "single" || !treeItem.selected;`

**Review by jcfranco — APPROVED** [review]
🎉

File: packages/calcite-components/src/components/tree/tree.tsx (hunk 1 above)
Comment (josercarcamo) [reply packages/calcite-components/src/components/tree/tree.tsx]: Actually, the two lines are not logically equivalent: The suggested line will return "true" whenever "selectionMode" is "single"; however, the existing line can return true or false when "selectionMode" is "single". The line can be changed to:

`this.selectionMode !== "single" || !treeItem.selected"`

Comment (driskull) [reply packages/calcite-components/src/components/tree/tree.tsx]: Changes look good. 👍 Merge approved

**Pull request merged**